server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1582,52 +1663,40 @@ private void executeOneStaleIndexDelete(
                 DeleteResult deleteResult = DeleteResult.ZERO;
                 try {
                     logger.debug("[{}] Found stale index [{}]. Cleaning it up", metadata.name(), indexSnId);
+                    boolean releaseLockOrTriggerCleanupFailed = false;
                     if (remoteStoreLockManagerFactory != null) {
                         final Map<String, BlobContainer> shardBlobs = indexEntry.getValue().children();
                         for (Map.Entry<String, BlobContainer> shardBlob : shardBlobs.entrySet()) {
                             for (String blob : shardBlob.getValue().listBlobs().keySet()) {
```

Comment:
we can replace `shardBlob.getValue().listBlobs().keySet()` with `shardBlob.getValue().listBlobsByPrefix(SHALLOW_SNAPSHOT_PREFIX)`